eth/core/processor.go

```diff
@@ -187,6 +187,7 @@ func (sp *StateProcessor) Finalize(
 
 // BuildAndRegisterPrecompiles builds the given precompiles and registers them with the precompile
 // plugin.
+// TODO: move precompile registration out of the state processor?
```

Comment:
think this is still necessary every block, as the default geth ones might change. Should probably delete the previous block's rules' default precompiles before registering the current block's rules' default precompiles to be exactly correct

Reply:
btw will have to add this in geth fork if removing our processor completely, cuz with our injected PrecompileManager system this is now necessary. before hand in upstream geth, it actually checks activePrecompiles every block to effectively get the same result